msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfigurationFactory.java

```diff
@@ -93,12 +99,27 @@ private static PublicClientApplicationConfiguration initializeConfigurationInter
             config.validateConfiguration();
         }
 
+        final PublicClientApplicationConfiguration configWithGlobal = mergeConfigurationWithGlobal(config);
+
         //Initialize internal library configuration
-        final LibraryConfiguration libraryConfiguration = LibraryConfiguration.builder().authorizationInCurrentTask((config.authorizationInCurrentTask())).build();
+        final LibraryConfiguration libraryConfiguration = LibraryConfiguration.builder().authorizationInCurrentTask((configWithGlobal.authorizationInCurrentTask())).build();
         LibraryConfiguration.intializeLibraryConfiguration(libraryConfiguration);
 
-        config.setOAuth2TokenCache(MsalOAuth2TokenCache.create(AndroidPlatformComponents.createFromContext(context)));
-        return config;
+        configWithGlobal.setOAuth2TokenCache(MsalOAuth2TokenCache.create(AndroidPlatformComponents.createFromContext(context)));
+        return configWithGlobal;
+    }
+
+    @WorkerThread
+    private static PublicClientApplicationConfiguration mergeConfigurationWithGlobal(final @NonNull PublicClientApplicationConfiguration developerConfig) {
+        if (!GlobalSettings.isGlobalSettingsInitialized()) {
+            Logger.warn(TAG + "mergeConfigurationWithGlobal",
+                    GlobalSettings.NO_GLOBAL_SETTINGS_WARNING);
+            return developerConfig;
+        }
+
+        developerConfig.mergeGlobalConfiguration(GlobalSettings.getGlobalSettingsConfiguration());
```

Comment:
suggestion: put `mergeGlobalConfiguration` (or this method) into `GlobalSettings`, and protect the method with a read lock. 

By doing this, we wouldn't have to expose the lock.